examples/user_guide/Dependencies_and_Watchers.ipynb

```diff
@@ -0,0 +1,820 @@
+    "        self.param.watch(self.run_a2, ['a'])\n",
+    "        self.param.watch(self.run_b,  ['b'])\n",
+    "\n",
+    "    def run_a1(self, event):\n",
+    "        self.b += 1\n",
+    "        print('a1', self.a, e(event))\n",
+    "\n",
+    "    def run_a2(self, event):\n",
+    "        print('a2', self.a, e(event))\n",
+    "\n",
+    "    def run_b(self, event):\n",
+    "        print('b', self.b, e(event))\n",
+    "        \n",
+    "p = P()\n",
+    "\n",
+    "p.a = 1"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "22f09fa4",
+   "metadata": {},
+   "source": [
+    "Here, we have set up three Watchers, each invoking a method on `P` when either `a` or `b` changes. The first Watcher invokes `run_a1` when `a` changes, and in turn `run_a1` changes `b`. Since `queued=True` for `run_a1`, `run_b` is not invoked while `run_a1` executes, but only later once both `run_a1` and `run_a2` have completed (since both Watchers were triggered by the original event `p.a=1`). \n",
+    "\n",
+    "Here we're using data from the `Event` objects given to each callback to see what's changed; try `help(param.parameterized.Event)` for details of what is in these objects (and similarly try the help for `Watcher` (returned by `watch`) or `PInfo` (returned by `.param.params_depended_on`))."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e6bfb4b5",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#help(param.parameterized.Event)\n",
+    "#help(param.parameterized.Watcher)\n",
+    "#help(param.parameterized.PInfo)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9d0dbaef",
+   "metadata": {},
+   "source": [
+    "## Using dependencies and watchers\n",
+    "\n",
+    "Whether you use the `watch` or the `depends` approach, Param will store a set of `Watcher` objects on each `Parameterized` object that let it manage and process `Event`s. Param provides various context managers, methods, and Parameters that help you work with Watchers and Events:\n",
+    "\n",
+    "- [`batch_call_watchers`](#batch_call_watchers): context manager accumulating and eliding multiple Events to be applied on exit from the context \n",
+    "- [`discard_events`](#discard_events): context manager silently discarding events generated while in the context\n",
+    "- [`.param.trigger`](#.param.trigger): method to force creation of an Event for this Parameter's Watchers without a corresponding change to the Parameter\n",
+    "- [Event Parameter](#Event-Parameter): Special Parameter type providing triggerable transient Events (like a momentary push button)\n",
+    "- [Async executor](#Async-executor): Support for asynchronous processing of Events, e.g. for interfacing to external servers\n",
+    "\n",
+    "Each of these will be described in the following sections."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "86823025",
+   "metadata": {},
+   "source": [
+    "### `batch_call_watchers`\n",
```

Comment:
Would be nice to mention `.param.set_param` here and that it batches by default.